client/hotrod-client/src/test/java/org/infinispan/client/hotrod/impl/transport/netty/CrashMidOperationTest.java

```diff
@@ -0,0 +1,139 @@
+package org.infinispan.client.hotrod.impl.transport.netty;
+
+import static org.infinispan.server.hotrod.test.HotRodTestingUtil.hotRodCacheConfiguration;
+import static org.testng.AssertJUnit.assertTrue;
+
+import java.net.ConnectException;
+import java.net.InetSocketAddress;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.infinispan.client.hotrod.DataFormat;
+import org.infinispan.client.hotrod.configuration.Configuration;
+import org.infinispan.client.hotrod.exceptions.TransportException;
+import org.infinispan.client.hotrod.impl.ClientTopology;
+import org.infinispan.client.hotrod.impl.operations.RetryOnFailureOperation;
+import org.infinispan.client.hotrod.retry.AbstractRetryTest;
+import org.infinispan.client.hotrod.test.HotRodClientTestingUtil;
+import org.infinispan.commons.test.Exceptions;
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.test.fwk.CleanupAfterMethod;
+import org.testng.annotations.Test;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.Channel;
+
+@CleanupAfterMethod
+@Test(testName = "client.hotrod.impl.transport.netty.CrashMidOperationTest", groups = "functional")
+public class CrashMidOperationTest extends AbstractRetryTest {
```

Comment:
This test might be superfluous.

It verifies that if the channel becomes inactive while executing an operation, the operation will fail, the subsequent operation will fail to establish a connection and include the server in the failed list, and the third operation finally succeeds.